### PR TITLE
Fix mongocxx r3.0.2 build error

### PR DESCRIPTION
**Pull request by adamchel**
Resolves issue #34 which came about due to a change in pointer types in the C++ driver (https://github.com/mongodb/mongo-cxx-driver/commit/4e49da57b359287d9aa70718d7b025f62fa21268).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,11 +84,11 @@ set(THIRD_PARTY_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/third_party)
 
 enable_testing()
 
-set(LIBBSONCXX_REQUIRED_VERSION 3.0.1)
+set(LIBBSONCXX_REQUIRED_VERSION 3.0.2)
```
Comment (Machyne) [inline CMakeLists.txt]: Should this be `3.1.1`?

File: src/boson/bson_streambuf.cpp [hunk 2]
```diff
@@ -57,8 +57,8 @@ int bson_output_streambuf::insert(int ch) {
         if (_len > BSON_MAX_SIZE) {
             throw std::invalid_argument("BSON document length is too large.");
         }
-        _data = std::unique_ptr<uint8_t, void (*)(std::uint8_t *)>(new uint8_t[_len],
-                                                                   [](uint8_t *p) { delete[] p; });
+        _data = std::unique_ptr<uint8_t[], void (*)(std::uint8_t *)>(
```
Comment (Machyne) [inline src/boson/bson_streambuf.cpp]: Does this compile with the default deleter type? i.e. can we construct a `document::value` from a `std::unique_ptr<uint8_t[]>`?

**Review by Machyne — APPROVED** [review]
=)

**Comment by acmorrow** [comment]
@adamchel - There have been some recent inquiries about getting this resolved:

https://stackoverflow.com/questions/45430300/which-mongodb-odm-for-c/45449656?noredirect=1#comment78580654_45449656

Can we try to get it moving again?

**Comment by rkargon** [comment]
I'll see if I can get Mangrove running on the latest version of the mongocxx driver, since it looks like we haven't really tried since February. 

In the meantime, it might be worth recommending that the StackOverflow user try to build Mangrove from Adam's fork and seeing if the problem persists.

**Pull request merged**